### PR TITLE
Improve GitHub workflow guidance

### LINKED ISSUE #479: Git remote setup
- Stop recommending gitflow
- Resources listed by @maurolepore  [common remote setups](https://happygitwithr.com/common-remote-setups.html) and [useful Git patterns for real life](https://happygitwithr.com/workflows-intro.html)

### PR DESCRIPTION
**Pull request by maelle**
Fix  #479

### CODE REVIEW

**Comment by maelle** [comment]
It seems we were already not recommending "git flow".

File: maintenance_collaboration.Rmd [hunk 1]
```diff
@@ -93,20 +93,27 @@ A possible model for onboarding collaborators is provided by Jim Hester in [his
...
 If your problem is _recruiting_ collaborators, you can post an open call like Jim Hester's [on Twitter](https://twitter.com/jimhester_/status/997109466674819074), [GitHub](https://github.com/jimhester/lintr/issues/318), and as an rOpenSci package author, you can ask for help in rOpenSci slack and ask rOpenSci team for ideas for recruiting new collaborators.
 
-### Working with collaborators (including yourself) {#gitflow}
+### Working with collaborators (including yourself) {#git-workflow}
 
 [Branches](https://happygitwithr.com/git-branches.html) are cheap. Use them extensively when developing features, testing out new ideas, fixing problems.
 
 One of the branches is the default / main branch, where, if you follow [trunk-based development](https://www.atlassian.com/continuous-delivery/continuous-integration/trunk-based-development), you "merge small, frequent updates".
 See also [GitHub flow](https://docs.github.com/en/get-started/quickstart/github-flow) and [GitLab flow](https://docs.gitlab.com/ee/topics/gitlab_flow.html) docs.
 You might want to pair the frequent incrementing of version numbers (in `DESCRIPTION`).
```
Comment (mpadge) [inline maintenance_collaboration.Rmd:102]:
```suggestion
You might want to frequently increment version numbers (in `DESCRIPTION`). 
```

**Review by mpadge — APPROVED** [review]
One tiny suggestion that you didn't even write here, but otherwise that's a fine improvement. Thanks!

**Comment by maelle** [comment]
Thanks @mpadge!

**Review by maurolepore — COMMENTED** [review]
LGTM. Thanks!

**Pull request closed without merging**